Assets/Scripts/NPCs/StateMachine/States/Chase.cs

```diff
@@ -54,11 +64,40 @@ private Vector3 ComputeChaseVelocity()
 
         if (Vector3.Magnitude(velocity) > _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity))
         {
-            velocity = (velocity / Vector3.Magnitude(velocity)) * _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity);
+            velocity = (velocity / Vector3.Magnitude(velocity)) *
+                       _npcManager.GetModelSpeed(NPCsGlobalVariables.ChaseMaxVelocity);
         }
 
         velocity.y = 0;
 
         return velocity;
     }
-}
+
+    private void ComputeAStar()
+    {
+        Vector3 nodeWithoutY = Vector3.zero;
+        if (_currentTargetLocation != NPCsGlobalVariables.DefaultInitialVector)
+        {
+            nodeWithoutY = new Vector3(_currentTargetLocation.x, 0, _currentTargetLocation.z);
+        }
+
+        if (_currentTargetLocation == NPCsGlobalVariables.DefaultInitialVector ||
+            Vector3.Distance(_npcManager.transform.position, nodeWithoutY) <
+            NPCsGlobalVariables.WithinTargetNodeRange)
+        {
+            Node npcNode = NavigationGraph.GetClosestNode(_npcManager.transform.position);
+            Node playerNode = NavigationGraph.GetClosestNode(_playerData.LastSeenPosition);
+
+            List<Node> path = Pathfinding.Astar(npcNode, playerNode);
+
+            if (path.Count < 2)
+            {
+                _currentTargetLocation = _playerData.LastSeenPosition;
+            }
+            else
+            {
+                _currentTargetLocation = path[1].position;
```

Comment:
Why path[1] and not path[0] ?

Reply:
Might be a good idea to use a variable instead of a magic number, though.

Reply:
done